initial commit for WorkspaceAnalysis module

scooperstein opened the pull request:
Preliminary commit for creation of the WorkspaceAnalysis module. Still more to develop here, but it would be great to get your feedback!

Thanks,
Stephane

capalmer85 commented on plugins/WorkspaceAnalysis.cc:
```diff
@@ -0,0 +1,82 @@
+    for(int i=0; i < (int)samples.size(); i++) {
+        std::cout<<"Initializing histograms..."<<std::endl;
+        const char* sname = samples[i].sampleName.c_str();
+        hists1D.push_back(new TH1F(sname, sname, 100, 0., 1.) );
```
Here you need to put a switch for filling which you'll fill in settings.txt.  You'll have to make some mapping between floats and models.  

modelDimension=1
#bkgmodel=hist_1d
#sigmodel=hist_1d
bkgmodel=1
sigmodel=1

Then in these lines:
if(*f["sigmodel"]==1 && sampleIndex<0) {
     models[sampleIndex]=TH1F(sname, sname, 100, 0., 1.);
}
if(*f["sigmodel"]==2 && sampleIndex<0) {
     models[sampleIndex]=TH2F(sname, sname, 100, 0., 1., 100, 0, 300);
}
... same for background.  Bkg and sig could have different models.

capalmer85 commented on plugins/WorkspaceAnalysis.cc:
```diff
@@ -0,0 +1,82 @@
+        std::cout<<i<<std::endl;
+        std::cout<<hists1D.size()<<std::endl;
+        hists1D[i]->Fill(BDT, weight);
+        hists2D[i]->Fill(BDT, H_mass_reg, weight);
```
Here you can look up modelDimension

capalmer85 commented:
I wrote a couple of comments.  You should also consider creating something
like RooContainer and making a map of these an object of
WorkspaceAnalysis.  E.g. in WorkspaceAnalysis.h:

std::map<int, RooContainer> models;  // from sampleIndex to RooContainer

On Mon, Feb 2, 2015 at 7:19 PM, scooperstein <notifications@github.com>
wrote:

> Preliminary commit for creation of the WorkspaceAnalysis module. Still
> more to develop here, but it would be great to get your feedback!
>
> Thanks,
> Stephane
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/capalmer85/PrincetonAnalysisTools/pull/12
> Commit Summary
>
>    - initial commit for WorkspaceAnalysis module
>
> File Changes
>
>    - *M* AnalysisManager.cc
>    <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12/files#diff-0>
>    (6)
>    - *M* HelperClasses/SampleContainer.cc
>    <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12/files#diff-1>
>    (6)
>    - *M* LinkDef.h
>    <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12/files#diff-2>
>    (1)
>    - *M* Makefile
>    <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12/files#diff-3>
>    (8)
>    - *A* WorkspaceAnalysis/RunAnalysis.py
>    <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12/files#diff-4>
>    (28)
>    - *A* WorkspaceAnalysis/cfg/existingbranches.txt
>    <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12/files#diff-5>
>    (9)
>    - *A* WorkspaceAnalysis/cfg/newbranches.txt
>    <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12/files#diff-6>
>    (1)
>    - *A* WorkspaceAnalysis/cfg/samples.txt
>    <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12/files#diff-7>
>    (1)
>    - *A* WorkspaceAnalysis/cfg/settings.txt
>    <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12/files#diff-8>
>    (22)
>    - *A* WorkspaceAnalysis/workspace_config.txt
>    <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12/files#diff-9>
>    (10)
>    - *A* plugins/WorkspaceAnalysis.cc
>    <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12/files#diff-10>
>    (82)
>    - *A* plugins/WorkspaceAnalysis.h
>    <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12/files#diff-11>
>    (41)
>
> Patch Links:
>
>    - https://github.com/capalmer85/PrincetonAnalysisTools/pull/12.patch
>    - https://github.com/capalmer85/PrincetonAnalysisTools/pull/12.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/capalmer85/PrincetonAnalysisTools/pull/12>.
>

scooperstein commented:
Hi Chris,

I have implemented a more functional version of WorkspaceAnalysis with the changes you mentioned. The new module takes as input the root file from VHbbAnalysis and outputs a root file with the workspace we need for running limits. It also outputs a root file with the 1D and 2D histograms for bookkeeping purposes.

You'll see that there are several new variables that need to be specified in settings. 'modelDimension' (1 or 2) says whether we want the 1D (BDT) or 2D (BDT vs Mjj_regressed) shapes in the workspace. The other variables specify the numbers of bins, min, and max values for the histograms. 

There were also a couple loose ends that I added to this commit from the addition of the regressions. I noticed while setting up locally that a couple things were missing/updated from the last commit. 

I still am not using the RooContainer class, that's something we'll still need to develop if we want to use it.

I'll see if I can squash this commit onto the first one without breaking everything!

Cheers,
Stephane

scooperstein commented:
Also the Makefile should now work with the new commits, which include RooFit/RooStats. As long as you ran ./config --enable-roofit when installing root it should work out of the box.
